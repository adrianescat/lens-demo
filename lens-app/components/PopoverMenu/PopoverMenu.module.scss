.popoverWrapper {
  position: relative;
  display: inline-block;

  .popoverContent {
    opacity: 0;
    visibility: hidden;
    position: absolute;
    left: -125px;
    transform: translate(0, 10px);
    background-color: #fff;
    box-shadow: 0 10px 17px 0px #f3f3f3;
    width: 160px;
    border-radius: 4px;
  }

  &:hover .popoverContent {
    z-index: 10;
    opacity: 1;
    visibility: visible;
    transform: translate(7px, 0);
    transition: all 0.5s cubic-bezier(0.75, -0.02, 0.2, 0.97);
  }

  .popoverLi {
    display: flex;
    justify-content: space-between;
    font-family: "MMedium";
    font-size: 12px;
    color: #313131;
    height: 40px;
    align-items: center;
    padding: 10px 20px;

    &:hover {
      background-color: #f6f6f6;
    }

    &.readOnly {
      color: #000;
    }

    &.readOnly:hover {
      background-color: #fff;

      .value {
        color: blue;
      }
    }
  }
}