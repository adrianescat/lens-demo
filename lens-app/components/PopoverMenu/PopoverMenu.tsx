import React, { ReactElement } from 'react'
import classnames from 'classnames';
import styles from './PopoverMenu.module.scss';

export interface Props {
  list: Array<any>;
  children: ReactElement;
}

export default function PopoverMenu({ list, children }: Props): ReactElement {
  return (
    <div className={styles.popoverWrapper}>
      {children}
      <div className={styles.popoverContent}>
        <ul>
          {list.map((option: any, index) => (
            <li key={`option-${index}`} className={classnames(styles.popoverLi, option.value ? styles.readOnly : '')}>
              <span>{option.label ? option.label : option}</span>
              {option.value && (
                <span className={styles.value}>{option.value}</span>
              )}
            </li>
          ))}
        </ul>
      </div>
    </div>
  )
}
