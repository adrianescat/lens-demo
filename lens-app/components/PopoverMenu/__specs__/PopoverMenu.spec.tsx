import React from 'react';
import { shallow } from 'enzyme';
import PopoverMenu, { Props } from '../PopoverMenu';

let wrapper: any;
let props: Props;

beforeEach(() => {
  props = {
    list: ['calendar', 'email'],
    children: <div>Test</div>,
  };
  wrapper = shallow(<PopoverMenu {...props} />);
});

test('should match snapshots', () => {
  expect(wrapper).toMatchSnapshot();
});

describe('when is read-only mode', () => {
  beforeEach(() => {
    props = {
      list: [{ level: 'label', value: 1 }, { level: 'label', value: 1 }],
      children: <div>Test</div>,
    };
    wrapper = shallow(<PopoverMenu {...props} />);
  });

  test('should match snapshots', () => {
    expect(wrapper).toMatchSnapshot();
  });
})
