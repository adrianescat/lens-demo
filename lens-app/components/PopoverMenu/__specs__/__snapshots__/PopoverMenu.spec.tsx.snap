// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshots 1`] = `
<div
  className="popoverWrapper"
>
  <div>
    Test
  </div>
  <div
    className="popoverContent"
  >
    <ul>
      <li
        className="popoverLi"
        key="option-0"
      >
        <span>
          calendar
        </span>
      </li>
      <li
        className="popoverLi"
        key="option-1"
      >
        <span>
          email
        </span>
      </li>
    </ul>
  </div>
</div>
`;

exports[`when is read-only mode should match snapshots 1`] = `
<div
  className="popoverWrapper"
>
  <div>
    Test
  </div>
  <div
    className="popoverContent"
  >
    <ul>
      <li
        className="popoverLi readOnly"
        key="option-0"
      >
        <span>
          <Component />
        </span>
        <span
          className="value"
        >
          1
        </span>
      </li>
      <li
        className="popoverLi readOnly"
        key="option-1"
      >
        <span>
          <Component />
        </span>
        <span
          className="value"
        >
          1
        </span>
      </li>
    </ul>
  </div>
</div>
`;
